crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -2960,6 +2961,101 @@ TEST_P(PBKDF2_ServiceIndicatorTest, PBKDF2) {
   EXPECT_EQ(approved, test.expect_approved);
 }
 
+// SSHKDF test vector from NIST CAVS 14.1 test vectors
+// https://csrc.nist.gov/Projects/Cryptographic-Algorithm-Validation-Program/Component-Testing#KDF135
+static const uint8_t kSSHKDFkey[] = {
+    0x00, 0x00, 0x00, 0x81, 0x00, 0x87, 0x5c, 0x55, 0x1c, 0xef, 0x52, 0x6a,
+    0x4a, 0x8b, 0xe1, 0xa7, 0xdf, 0x27, 0xe9, 0xed, 0x35, 0x4b, 0xac, 0x9a,
+    0xfb, 0x71, 0xf5, 0x3d, 0xba, 0xe9, 0x05, 0x67, 0x9d, 0x14, 0xf9, 0xfa,
+    0xf2, 0x46, 0x9c, 0x53, 0x45, 0x7c, 0xf8, 0x0a, 0x36, 0x6b, 0xe2, 0x78,
+    0x96, 0x5b, 0xa6, 0x25, 0x52, 0x76, 0xca, 0x2d, 0x9f, 0x4a, 0x97, 0xd2,
+    0x71, 0xf7, 0x1e, 0x50, 0xd8, 0xa9, 0xec, 0x46, 0x25, 0x3a, 0x6a, 0x90,
+    0x6a, 0xc2, 0xc5, 0xe4, 0xf4, 0x8b, 0x27, 0xa6, 0x3c, 0xe0, 0x8d, 0x80,
+    0x39, 0x0a, 0x49, 0x2a, 0xa4, 0x3b, 0xad, 0x9d, 0x88, 0x2c, 0xca, 0xc2,
+    0x3d, 0xac, 0x88, 0xbc, 0xad, 0xa4, 0xb4, 0xd4, 0x26, 0xa3, 0x62, 0x08,
+    0x3d, 0xab, 0x65, 0x69, 0xc5, 0x4c, 0x22, 0x4d, 0xd2, 0xd8, 0x76, 0x43,
+    0xaa, 0x22, 0x76, 0x93, 0xe1, 0x41, 0xad, 0x16, 0x30, 0xce, 0x13, 0x14,
+    0x4e
+};
+
+static const uint8_t kSSHKDFxcghash[] = {
+    0x0e, 0x68, 0x3f, 0xc8, 0xa9, 0xed, 0x7c, 0x2f, 0xf0, 0x2d, 0xef, 0x23,
+    0xb2, 0x74, 0x5e, 0xbc, 0x99, 0xb2, 0x67, 0xda, 0xa8, 0x6a, 0x4a, 0xa7,
+    0x69, 0x72, 0x39, 0x08, 0x82, 0x53, 0xf6, 0x42
+};
+
+static const uint8_t kSSHKDFsession[] = {
+    0x0e, 0x68, 0x3f, 0xc8, 0xa9, 0xed, 0x7c, 0x2f, 0xf0, 0x2d, 0xef, 0x23,
+    0xb2, 0x74, 0x5e, 0xbc, 0x99, 0xb2, 0x67, 0xda, 0xa8, 0x6a, 0x4a, 0xa7,
+    0x69, 0x72, 0x39, 0x08, 0x82, 0x53, 0xf6, 0x42
+};
+
+static const char kSSHKDFtype = EVP_KDF_SSHKDF_TYPE_INITIAL_IV_CLI_TO_SRV;
+
+static const uint8_t kSSHKDFexpected[] = {
+    0x41, 0xff, 0x2e, 0xad, 0x16, 0x83, 0xf1, 0xe6
+};
+
+static const struct SSHKDFTestVector {
+    // func is the hash function for PBKDF2 to test.
+    const EVP_MD *(*func)();
+    const uint8_t *key;
+    const size_t key_len;
+    const uint8_t *xcghash;
+    const size_t xcghash_len;
+    const uint8_t *session_id;
+    const size_t session_id_len;
+    const char type;
+    const size_t output_len;
+    const uint8_t *expected_output;
+    const FIPSStatus expect_approved;
+} kSSHKDFTestVectors[] = {
+    {
+        EVP_sha256,
```

Comment:
I understand our trend in the service indicator is to test all possible inputs that are approved and one non-approved (or more, if there are different combinations). So I suggest adding tests for the other approved MDs, with one type of key, not all types as in the unit tests.